src/main/java/de/medizininformatik_initiative/process/data_transfer/service/InsertData.java

```diff
@@ -119,10 +125,20 @@ private List<IdType> storeData(FhirClient fhirClient, Bundle bundle, String send
 		return idsOfCreatedResources;
 	}
 
-	private Bundle checkAndAdaptBundleForExistingData(FhirClient fhirClient, Bundle bundle, String sendingOrganization,
-			String projectIdentifier, Task task)
+	private IdType createBinary(Binary binary)
```

Comment:
Since the Binary resource is extracted from a Bundle resource that was stored in a BPMN variable. The Binary resource will have been serialized between BPMN tasks. Using a stream based create method here, removes one serialization step, but may not solve #34.

Reply:
Implemented in 94db4a5 and 72adeee